src/plugins/sockets/inGameEvents.js

```diff
@@ -30,10 +30,11 @@ export async function handleInGameEvents(evData) {
     }
     case SocketEvent.INITIALIZE: {
       // Update state
-      gameStore.resetPNumIfNullThenUpdateGame(evData.game);
       if (isSpectating) {
         gameStore.myPNum = 0; // always spectate as p0
+        gameStore.isSpectating = true;
```

Comment:
I understand why we’re setting this, but it feels weird to me to consider a normal player a spectator. Is this initialize function used anywhere else or just in the main game start up logic?

Reply:
I'm pretty sure it's called at the main game startup, I kind of put it here as a backup I'm not sure it's 100% needed because I set it in requestSubscribe,

Reply:
I think it might be better to read `isSpectating` directly off the currentRoute in the router so that we don't need to set it in multiple different places. That way we have a centralized source of truth for isSpectating (the route) and we don't need to worry about keeping them in sync.

Reply:
Would it make sense to keep this variable in the store, but set it to true in the `beforeEnter` in the router? And then we can change `isSpectating` in `GameView` to `gameStore.isSpectating`